File: Assignment 4 - 2D Arrays (Deadline 29 June 2023 )/arrayPairSum.js
function arrayPairSum(nums) {
    nums.sort((a, b) => a - b); // Sort the array in ascending order
    let sum = 0;
  
    for (let i = 0; i < nums.length; i += 2) {
      sum += nums[i];
    }
  
    return sum;
  }
  
  const nums = [1, 4, 3, 2];
  const maximizedSum = arrayPairSum(nums);
  console.log(maximizedSum);  

File: Assignment 2 - Arrays (Deadline  26 June 2023 )/ArrayPairSum.java
import java.util.Arrays;

public class ArrayPairSum {
    public int arrayPairSum(int[] nums) {
        // Sort the array in ascending order
        Arrays.sort(nums);

        int sum = 0;
        // Iterate over the sorted array, skipping every second element
        for (int i = 0; i < nums.length; i += 2) {
            // Add the smaller element of each pair to the sum
            sum += nums[i];
        }

        return sum;
    }

    public static void main(String[] args) {
        int[] nums = {1, 4, 3, 2};
        ArrayPairSum solution = new ArrayPairSum();
        int maxSum = solution.arrayPairSum(nums);
        System.out.println("Maximum sum: " + maxSum);
    }
}
